crates/derive/src/stages/attributes_queue/builder.rs

```diff
@@ -0,0 +1,167 @@
+            if l2_parent.l1_origin.hash != info.parent_hash {
+                return Err(BuilderError::BlockMismatchEpochReset(
+                    epoch,
+                    l2_parent.l1_origin,
+                    info.parent_hash,
+                ));
+            }
+            let receipts = self.receipts_fetcher.receipts_by_hash(epoch.hash).await?;
+            sys_config.update_with_receipts(&receipts, &self.rollup_cfg, info.timestamp)?;
+            let deposits =
+                derive_deposits(epoch.hash, receipts, self.rollup_cfg.deposit_contract_address)
+                    .await?;
+            l1_info = info;
+            deposit_transactions = deposits;
+            // sequence_number = 0;
+        } else {
+            #[allow(clippy::collapsible_else_if)]
+            if l2_parent.l1_origin.hash != epoch.hash {
+                return Err(BuilderError::BlockMismatch(epoch, l2_parent.l1_origin));
+            }
+
+            let info = self.receipts_fetcher.info_by_hash(epoch.hash).await?;
+            l1_info = info;
+            deposit_transactions = vec![];
+            // sequence_number = l2_parent.seq_num + 1;
+        }
+
+        // Sanity check the L1 origin was correctly selected to maintain the time invariant
+        // between L1 and L2.
+        let next_l2_time = l2_parent.block_info.timestamp + self.rollup_cfg.block_time;
+        if next_l2_time < l1_info.timestamp {
+            return Err(BuilderError::BrokenTimeInvariant(
+                l2_parent.l1_origin,
+                next_l2_time,
+                l1_info.id(),
+                l1_info.timestamp,
+            ));
+        }
+
+        let mut upgrade_transactions: Vec<RawTransaction> = vec![];
+        if self.rollup_cfg.is_ecotone_active(next_l2_time) {
+            upgrade_transactions =
+                EcotoneTransactionBuilder::build_txs().map_err(BuilderError::Custom)?;
+        }
+
+        // let l1_info_tx = l1_info_deposit_bytes(self.rollup_cfg, sys_config, sequence_number,
+        // l1_info, next_l2_time)?;
+
+        let mut txs = vec![];
+        // txs.push(l1_info_tx);
+        txs.extend(deposit_transactions);
+        txs.extend(upgrade_transactions);
+
+        let withdrawals = None;
+        if self.rollup_cfg.is_canyon_active(next_l2_time) {
+            // withdrawals = Some(Withdrawals::default());
+        }
+
+        let parent_beacon_root = None;
+        if self.rollup_cfg.is_ecotone_active(next_l2_time) {
+            // parent_beacon_root = Some(l1_info.parent_beacon_root().unwrap_or_default());
+            // if the parent beacon root is not available, default to zero hash
+        }
+
+        Ok(PayloadAttributes {
+            timestamp: next_l2_time,
+            // TODO: The mix digest is pulled from the l1 info, which is a **full** block.
+            //       Currently, the l1_info is only the minimal `BlockInfo` defined in our types.
+            prev_randao: B256::default(),
```

Comment:
Ah yeah we'll need this. I think we only need to have the header here?

Reply:
See [review comment](https://github.com/ethereum-optimism/kona/pull/92#pullrequestreview-1999686310)